.github/workflows/benchmark-local.yaml

```diff
@@ -0,0 +1,73 @@
+# cli command to run this from repo
+# given a run, get output (link?)
+# input variable to CLI to allow users to specify where to output their artefacts
+
+name: Run performance comparisons
+
+on:
+  workflow_dispatch:
+    inputs:
+      c1:
+        description: First commit hash
+        required: true
+      c2:
+        description: Second commit hash
+        required: true
+
+jobs:
+  build:
+    runs-on: [self-hosted, linux, x64, bench]
+    strategy:
+      fail-fast: true
+      matrix:
+        commit: ["${{ github.event.inputs.c1 }}", "${{ github.event.inputs.c2 }}"]
+    timeout-minutes: 15
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+    - name: Assume AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: us-west-2
+        role-to-assume: ${{ secrets.ACTIONS_AWS_ROLE_ARN }}
+        role-session-name: daft-performance-comparisons
+
+    - name: "Checkout commit ${{ matrix.commit }}"
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ matrix.commit }}
+
+    - name: Install uv and python
+      run: |
+        curl -LsSf https://astral.sh/uv/install.sh | sh
+        source $HOME/.cargo/env
+        uv python install 3.9
+        uv python pin 3.9
+
+    - name: Mount NVMe drive
+      run: |
+        sudo mkfs.ext4 /dev/nvme0n1
+        sudo mount -t ext4 /dev/nvme0n1 /tmp
+        sudo chmod 777 /tmp
+
+    - name: Download all pre-requisites dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install gcc g++ pkg-config libssl-dev -y
+
+    - name: "Build wheel for commit ${{ matrix.commit }}"
+      run: |
+        # instantiate venv and run build
+        export CARGO_TARGET_DIR=/tmp/target
+        uv v
+        source .venv/bin/activate
+        uv pip install pip
+        uv pip install maturin
+        maturin build
+
+    - name: Upload files to s3
+      run: |
+        for file in /tmp/target/wheels/*.whl; do
+          aws s3 cp $file s3://github-actions-artifacts-bucket/daft-performance-comparisons/${{ github.sha }}/${{ matrix.commit }} --acl public-read --no-progress;
+        done
```

Comment:
After this is uploaded, does the index html get updated? You should check to see how the other S3 artifacts workflow does it.

Reply:
This is not updating the index html. I can look into updating it.

The only thing is, would we want to update it? These are all going to be internally used, and downloading the wheel directly from the link rather than having to add an index to the `trusted-indexes` and then performing the whole indexing logic seems a little unnecessary? I'm not sure, maybe there's a reason to do this.

Reply:
Oh this was a question more so than pointing out a problem.

I see what you mean though. If we are able to download the wheel directly from a link then the index won't be necessary.